### ISSUE TITLE
Master can't find kernels when building from binary, returns INVALID_KERNEL_NAME

### ISSUE THREAD

**Issue by inducer** [pocl/pocl#491]
Running this

```
src = """
__kernel void __attribute__ ((reqd_work_group_size(1, 1, 1))) loopy_kernel(int const n, __global int *__restrict__ v, __global double *__restrict__ z, int const i, int const itrip)
{
}

__kernel void __attribute__ ((reqd_work_group_size(1, 1, 1))) loopy_kernel_1(int const n, __global int *__restrict__ v, __global double *__restrict__ z, int const i)
{
}

"""

import pyopencl as cl
import numpy as np

ctx = cl.create_some_context()
queue = cl.CommandQueue(ctx)
prg = cl.Program(ctx, src).build()
assert prg.loopy_kernel_1.num_args == 4

binary = prg.binaries[0]
prg2 = cl.Program(ctx, [ctx.devices[0]], [binary]).build()

assert prg2.loopy_kernel_1.num_args == 4
```
on 06432cdb (current `master` or close to it) with a patched (#1) LLVM 4.0.1 yields
```
/home/andreask_work/src/pytools/pytools/py_codegen.py:136: PendingDeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Traceback (most recent call last):
  File "/home/andreask_work/src/pyopencl/pyopencl/__init__.py", line 320, in __getattr__
    knl = Kernel(self, attr)
  File "/home/andreask_work/src/pyopencl/pyopencl/cffi_cl.py", line 1687, in __init__
    _to_cstring(name)))
  File "/home/andreask_work/src/pyopencl/pyopencl/cffi_cl.py", line 645, in _handle_error
    raise e
pyopencl.cffi_cl.LogicError: clCreateKernel failed: INVALID_KERNEL_NAME

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "binary-retrieval.py", line 23, in <module>
    assert prg2.loopy_kernel_1.num_args == 4
  File "/home/andreask_work/src/pyopencl/pyopencl/__init__.py", line 335, in __getattr__
    "info attribute or as a kernel name" % attr)
AttributeError: 'loopy_kernel_1' was not found as a program info attribute or as a kernel name
```

**Comment by franz** [comment]
trying that program on 06432cd (current master), i've run into an entirely different bug, after a small workaround i cannot reproduce this bug (IOW it exits with status 0).

**Comment by pjaaskel** [comment]
@induce do you happen to have an old binary you try to load? The format was updated and might not be backwards compatible.

**Comment by franz** [comment]
@pjaaskel @inducer managed to reproduce it. It was that "entirely different" bug i mentioned, it just failed  elsewhere (much sooner) for me because i have pocl built with assertions. Should be fixed with next PR

**Comment by franz** [comment]
This should now be fixed in git master, @inducer  please retest

**Comment by inducer** [comment]
Confirmed fixed in 07ac4415. Additionally, the entire PyOpenCL test suite passes with that revision. Thanks for working on this!
